### PR TITLE
Moved THTensor_(exponential) to ATen

### PR DESCRIPTION
**Pull request by moon05**
porting exponential cpu from ATen

### CODE REVIEW

**Review by syed-ahmed — APPROVED** [review]

**Review by zou3519 — CHANGES_REQUESTED** [review]
Could we see some before/after performance numbers to ensure there isn't a performance regression?

File: aten/src/ATen/native/Distributions.cpp [hunk 1]
```diff
@@ -172,6 +172,18 @@ Tensor& bernoulli_scalar_cpu_(Tensor& self, double p, Generator* gen) {
 }
 
 
+Tensor& exponential_cpu_(Tensor& self, double lambda, Generator* gen) {
+  AT_DISPATCH_ALL_TYPES(self.scalar_type(), "exponential_cpu_", [&] {
+    THGenerator* generator = get_generator(gen);
+    std::lock_guard<std::mutex> lock(generator->mutex);
+    CPU_tensor_apply1<scalar_t>(
```
Comment (fmassa) [inline aten/src/ATen/native/Distributions.cpp]: Did we merge already the PR that adds support for bernoulli with TensorIterator?

Comment (VitalyFedyunin) [reply aten/src/ATen/native/Distributions.cpp]: It was done for CUDA but it is applicable for CPU https://github.com/pytorch/pytorch/pull/21300/files

**Comment by VitalyFedyunin** [comment]
Implements #24699

**Comment by VitalyFedyunin** [comment]
Needs rebase. Also still waiting for benchmarks results.

**Review by VitalyFedyunin — CHANGES_REQUESTED** [review]
Needs rebase. Also still waiting for benchmarks results.

**Pull request closed without merging**